data/src/main/java/com/linkedin/data/schema/PdlBuilder.java

```diff
@@ -127,14 +129,17 @@ PdlBuilder writeSpace() throws IOException
   abstract boolean writeDoc(String doc) throws IOException;
 
   /**
-   * Writes a set of schema properties that share a common prefix to .pdl.
+   * Writes a set of schema properties that share a common prefix to .pdl. Sorts the properties by key before writing.
    *
    * @param prefix provides the common prefix of all the properties.
    * @param properties provides the properties to write.
    */
   PdlBuilder writeProperties(List<String> prefix, Map<String, Object> properties) throws IOException
   {
-    for (Map.Entry<String, Object> entry : properties.entrySet())
+    for (Map.Entry<String, Object> entry :
+        properties.entrySet().stream()
```

Comment:
I'd avoid packing this stream builder chain into the for-loop statement. Either construct the list on its own line, or use `Stream#forEach`

Reply:
fixed